layers/+lang/perl6/packages.el

```diff
@@ -26,26 +26,24 @@
     company
     evil
     flycheck
-    (flycheck-perl6 :requires flycheck)
+    (flycheck-raku :requires flycheck)
     raku-mode
     ))
 
 (defun perl6/post-init-company ()
   (spacemacs|add-company-backends
     :backends company-capf
-    :modes perl6-mode))
+    :modes raku-mode))
 
 (defun perl6/post-init-evil ()
-  (add-to-list 'spacemacs-jump-handlers-perl6-mode 'evil-jump-to-tag))
+  (add-to-list 'spacemacs-jump-handlers-raku-mode 'evil-jump-to-tag))
 
 (defun perl6/post-init-flycheck ()
-  (spacemacs/enable-flycheck 'perl6-mode))
+  (spacemacs/enable-flycheck 'raku-mode))
 
-(defun perl6/init-flycheck-perl6 ()
-  (with-eval-after-load 'flycheck
-    (require 'flycheck-perl6)))
+(defun perl6/init-flycheck-raku ()
+  (add-hook 'flycheck-mode-hook (lambda () (use-package flycheck-raku))))
```

Comment:
Unnecessary to `add-hook`
Just write
```elisp
(use-package flycheck-raku
  :defer t)
```

Reply:
Thank you for the suggestion! Can you help me understand how the package will be loaded in this case? "Defer" will prevent it from loading on start and there are no autoloads set up for it.

Reply:
okay I see